### DISCUSSION TITLE
How to avoid updates when "ref" property is changed?

### DISCUSSION THREAD

**Discussion by saivanji** [pmndrs/valtio#677] [Q&A]
In the following code, a change on the `bar` property triggers the update:
```ts
const state = proxy({
  foo: 1,
  bar: ref({
    baz: 2
  })
})
```

It seems, `ref` is used when we need to ignore the updates inside the object (like, `{ baz: 2}`).

But how can we ignore the update when the `bar` property changes?

**Comment by dai-shi** [comment]
Only the solution I came up with instantly is to nest it.
```js
const state = proxy({
  foo: 1,
  barRef: ref({
    bar: {
      baz: 2
    }
  })
})
```
No magic, right?

**Reply by saivanji** [reply]
Yeah, but the requirement is to retain the initial object structure.

So far the solution that works for me is:

```ts
const [proxy] = unstable_buildProxyFunction(undefined, (target, handler) => {
  const set = handler.set?.bind(handler)
  handler.set = (target, p, value, receiver) => {
    if (p !== 'bar') {
      target[p as keyof typeof target] = value
      return true
    }
    return set?.(target, p, value, receiver) || true
  }
  return new Proxy(target, handler)
})
```

**Reply by dai-shi** [reply]
Or you can double proxy, instead of `unstable_buildProxyFunction`, which is not unstable.

```js
import { proxy as origProxy } from 'valtio'

export const proxy = (obj) => new Proxy(origProxy(obj), {
  set: ...
})
```
